src/Runtime/Distributions/DiscreteChar.cs

```diff
@@ -1971,15 +2057,23 @@ public void SortAndCheckRanges()
             /// <summary>
             /// Normalizes probabilities in ranges and returns build Storage.
             /// </summary>
-            public Storage GetResult()
+            public Storage GetResult(double? maximumProbability = null)
             {
                 this.MergeNeighboringRanges();
-                this.NormalizeProbabilities();
-                return Storage.Create(
-                    this.ranges.ToArray(),
-                    this.charClasses,
-                    this.regexRepresentation,
-                    this.symbolRepresentation);
+                NormalizeProbabilities(this.ranges, maximumProbability);
+                return
+                    maximumProbability.HasValue
+                        ? Storage.CreateUncached(
```

Comment:
This would be cleaner if Storage.Create() just had a bool for cached/uncached.  But I appreciate you may want a separate call to ensure it is not accidentally used.

Reply:
Yes, I think that it is good to be explicit. But actually, I was just following the existing code. So I won't change this.